boa/src/value/mod.rs

```diff
@@ -713,10 +713,10 @@ impl Value {
                     .expect("String was not initialized")
                     .get_field(PROTOTYPE);
 
-                Ok(GcObject::new(Object::with_prototype(
-                    proto,
-                    ObjectData::String(string.clone()),
-                )))
+                let mut obj = Object::with_prototype(proto, ObjectData::String(string.clone()));
+                // Make sure the correct length is set on our new string object
+                obj.set(PropertyKey::from("length"), string.chars().count().into());
```

Comment:
```suggestion
                obj.set("length".into(), string.chars().count().into());
```
It's cleaner if we use `.into()` instead of `from`